dojo/finding/views.py

```diff
@@ -294,34 +307,50 @@ def close_finding(request, fid):
     finding = get_object_or_404(Finding, id=fid)
     # in order to close a finding, we need to capture why it was closed
     # we can do this with a Note
+    note_type_activation = Note_Type.objects.filter(is_active=True)
+    if len(note_type_activation):
+        missing_note_types = get_missing_mandatory_notetypes(finding)
+    else:
+        missing_note_types = note_type_activation
+    form = CloseFindingForm(missing_note_types=missing_note_types)
     if request.method == 'POST':
-        form = CloseFindingForm(request.POST)
-
+        form = CloseFindingForm(request.POST, missing_note_types=missing_note_types)
         if form.is_valid():
             now = timezone.now()
             new_note = form.save(commit=False)
             new_note.author = request.user
             new_note.date = now
             new_note.save()
             finding.notes.add(new_note)
-            finding.active = False
-            finding.mitigated = now
-            finding.mitigated_by = request.user
-            finding.last_reviewed = finding.mitigated
-            finding.last_reviewed_by = request.user
-            finding.endpoints.clear()
-            finding.save()
-
             messages.add_message(
                 request,
                 messages.SUCCESS,
-                'Finding closed.',
+                'Note Saved.',
```

Comment:
Is this change correct? This is "close finding" view it.

Reply:
Saving a note and closing a finding are separated with a condition. If a finding satisfies the condition it will be closed and "Finding closed" message will be displayed only after that.